util/internal/test/common/src/main/java/org/hibernate/search/util/impl/test/extension/parameterized/ParameterizedClassPostDiscoveryFilter.java

```diff
@@ -28,6 +31,19 @@
 public class ParameterizedClassPostDiscoveryFilter implements org.junit.platform.launcher.PostDiscoveryFilter {
 	@Override
 	public FilterResult apply(TestDescriptor testDescriptor) {
+		if ( testDescriptor instanceof ClassBasedTestDescriptor ) {
+			Class<?> testClass = ( (ClassBasedTestDescriptor) testDescriptor ).getTestClass();
+			// Not really related to parameterized tests but to not create another filter just for it ...
+			// We want to double-check that we don't have any static classes with @Nested annotation:
+			for ( Class<?> inner : testClass.getDeclaredClasses() ) {
+				if ( Modifier.isStatic( inner.getModifiers() ) && isAnnotated( inner, Nested.class ) ) {
+					throw new IllegalStateException(
+							"Nested test classes annotated with @Nested cannot be static, as they won't be executed. "
+									+ "Make " + inner + " non-static or remove the annotation (@Nested)." );
+				}
+			}
+		}
```

Comment:
Since this isn't related to `@ParameterizedSetup`, it would probably be best to move it to a separate filter?